registrations/management/commands/fix_registration_subscriptions.py

```diff
@@ -0,0 +1,351 @@
+                utils.post_message(payload)
+
+        request = SubscriptionRequest.objects.create(**sub)
+        post_save.connect(receiver=model_saved,
+                          dispatch_uid='instance-saved-hook')
+        return request
+
+    def sub_request_difference(self, request, expected):
+        expected_copy = [(key, expected[key])
+                         for key in sorted(expected.keys())]
+        actual = [(key, getattr(request, key))
+                  for key in sorted(expected.keys())]
+        return (expected_copy, actual, expected_copy != actual)
+
+    def verify_subscription(self, client, sub_request, subscriptions,
+                            apply_fix=False):
+        """
+        Verify a list of subscriptions based on a given subscription request.
+
+        :param sbm_client StageBasedMessagingApiClient:
+            The client to the stage based messaging service API
+        :param sub_request SubscriptionRequest:
+            The subscription request to use for the verification
+        :param subscriptions list:
+            The subscriptions to verify
+        :param apply_fix bool:
+            Whether or not to apply the fix or just log it.
+        """
+        msgset_subscription = {}
+        if subscriptions:
+            for sub in subscriptions[sub_request.identity]:
+                if sub_request.messageset == sub["messageset"]:
+                    msgset_subscription = sub
+                    # Prefer active subscriptions
+                    if sub["active"]:
+                        break
+        if not msgset_subscription:
+            self.log('No subscription found for subscription request %s' %
+                     (sub_request.id))
+            if apply_fix:
+                self.fire_subscription_hooks(sub_request)
+            return
+
+        # Don't do anything if there aren't active subscriptions
+        if not msgset_subscription["active"]:
+            self.log('No active subscription found for subscription request '
+                     '%s. Taking no action' % (sub_request.id))
+            return
+
+        # Don't do anything if the subscription is up to date
+        if sub_request.next_sequence_number == \
+                msgset_subscription["next_sequence_number"]:
+            self.log('Next sequence numbers match. Taking no action')
+            return
+
+        self.log('Subscription %s has next_sequence_number=%s, should be %s' %
+                 (msgset_subscription['id'],
+                  msgset_subscription["next_sequence_number"],
+                  sub_request.next_sequence_number))
+        if apply_fix:
+            client.update_subscription(
+                msgset_subscription['id'],
+                {'next_sequence_number': sub_request.next_sequence_number})
+            self.log('Updated subscription %s. Set next_sequence_number to %s'
+                     % (msgset_subscription['id'],
+                        sub_request.next_sequence_number))
```

Comment:
I'm a little confused by this bit.

If the subscription has just been created, then we should expect the `next_sequence_number` on the subscription request to match the actual subscription.

But, if the subscription has run even once, then we should expect the two `next_sequence_number`s to not match. But it seems here for that case, we will be resetting the user to the beginning of the messageset again.

Should we not be comparing the subscriptionrequest's `next_sequence_number + calculated_number_of_runs` to the subscription's `next_sequence_number`?